src/File/Path/Filename/DefaultTrait.php

```diff
@@ -16,7 +16,7 @@ public function filename()
     {
         $processor = Hash::get($this->settings, 'nameCallback', null);
         if (is_callable($processor)) {
-            return $processor($this->data, $this->settings);
+            return $processor($this->table, $this->entity, $this->data, $this->field, $this->settings);
```

Comment:
this is not BC

Reply:
What do you mean with "BC"?

Reply:
https://book.cakephp.org/3.0/en/contributing/backwards-compatibility.html